pandas/api/exchange/dataframe_protocol.py

```diff
@@ -0,0 +1,456 @@
+from typing import Tuple, Optional, Dict, Any, Iterable, Sequence, TypedDict
+import enum
+from abc import ABC, abstractmethod
+
```

Comment:
don't put this here, instead just import the public methods / properties.

locate in

pandas.core.exchange i think

Reply:
this file is the spec, expected to be a verbatim copy from the https://github.com/data-apis/dataframe-api - I'll add the comment in the file explaining that.